File: Codes/2600-k-items-with-the-maximum-sum.cpp
/*
 * @Author: LetMeFly
 * @Date: 2023-07-05 07:30:48
 * @LastEditors: LetMeFly
 * @LastEditTime: 2023-07-05 07:32:11
 */
#ifdef _WIN32
#include "_[1,2]toVector.h"
#endif

class Solution {
public:
    int kItemsWithMaximumSum(int numOnes, int numZeros, int numNegOnes, int k) {
        int ans = 0;
        int a[3][2] = {{numOnes, 1}, {numZeros, 0}, {numNegOnes, -1}};
        for (int i = 0; i < 3; i++) {
            int thisNum = min(k, a[i][0]);
            ans += a[i][1] * thisNum;
            k -= thisNum;
        }
        return ans;
    }
};

File: Codes/2600-k-items-with-the-maximum-sum.py
'''
Author: LetMeFly
Date: 2023-07-05 07:30:55
LastEditors: LetMeFly
LastEditTime: 2023-07-05 07:43:45
'''
class Solution:
    def kItemsWithMaximumSum(self, numOnes: int, numZeros: int, numNegOnes: int, k: int) -> int:
        ans = 0
        a = [[numOnes, 1], [numZeros, 0], [numNegOnes, -1]]
        for i in range(3):
            thisNum = min(k, a[i][0])
            ans += thisNum * a[i][1]
            k -= thisNum
        return ans
